tests/d3d12_device.c

```diff
@@ -356,13 +559,75 @@ void test_format_support(void)
     ok(!format_support.Support2 || format_support.Support2 == D3D12_FORMAT_SUPPORT2_TILED,
             "Got unexpected support2 %#x.\n", format_support.Support2);
 
-    for (i = 0; i < ARRAY_SIZE(depth_stencil_formats); ++i)
+    memset(&feature_levels, 0, sizeof(feature_levels));
+    feature_levels.NumFeatureLevels = ARRAY_SIZE(all_feature_levels);
+    feature_levels.pFeatureLevelsRequested = all_feature_levels;
+    hr = ID3D12Device_CheckFeatureSupport(device, D3D12_FEATURE_FEATURE_LEVELS, 
+        &feature_levels, sizeof(feature_levels));
+    ok(hr == S_OK, "Got unexpected hr %#x.\n", hr);
+
+    for (k = 0; k < ARRAY_SIZE(dxgi_format_list); ++k)
+    {
+        if (dxgi_format_list[k].feature_level == feature_levels.MaxSupportedFeatureLevel)
+            break;
+    }
+
+    for (i = 0; i < ARRAY_SIZE(all_formats); ++i)
     {
+        required_but_fails = false;
+        unspecified_format = false;
+        optional_format = false;
+
+        for (j = 0; j < ARRAY_SIZE(known_required_but_fails); ++j)
+        {
+            if (known_required_but_fails[j] == all_formats[i])
+            {
+                required_but_fails = true;
```

Comment:
Is this vkd3d-proton that fails, but not native drivers? In case, that should be marked with a todo_if() in the check, and not skipped.

Reply:
Marked